haystack/nodes/prompt/invocation_layer/sagemaker.py

```diff
@@ -0,0 +1,221 @@
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                aws_session_token=aws_session_token,
+                region_name=region_name,
+                profile_name=profile_name,
+            )
+            self.client = session.client("sagemaker-runtime")
+        except Exception as e:
+            raise SageMakerInferenceError(
+                f"Could not connect to SageMaker Inference Endpoint {model_name_or_path}."
+                f"Make sure the Endpoint exists and AWS environment is configured. {e}"
+            )
+        self.max_length = max_length
+
+        # for a list of supported parameters (TODO: verify if all of them are supported in sagemaker)
+        self.model_input_kwargs = {
+            key: kwargs[key]
+            for key in [
+                "best_of",
+                "details",
+                "do_sample",
+                "max_new_tokens",
+                "max_time",
+                "model_max_length",
+                "num_return_sequences",
+                "repetition_penalty",
+                "return_full_text",
+                "seed",
+                "temperature",
+                "top_k",
+                "top_p",
+                "truncate",
+                "typical_p",
+                "watermark",
+            ]
+            if key in kwargs
+        }
+
+        # We pop the model_max_length from the model_input_kwargs as it is not sent to the model
+        # but used to truncate the prompt if needed
+        model_max_length = self.model_input_kwargs.pop("model_max_length", 1024)
+
+        # Truncate prompt if prompt tokens > model_max_length-max_length (max_lengt is the length of the generated text)
+        self.prompt_handler = DefaultPromptHandler(
+            model_name_or_path="gpt2", model_max_length=model_max_length, max_length=self.max_length or 100
+        )
+
+    def invoke(self, *args, **kwargs) -> List[str]:
+        """
+        Sends the prompt to the remote model and returns the generated response(s).
+        You can pass all parameters supported by the Huggingface Transformers `generate` method
+        here via **kwargs (e.g. "temperature", "stop" ...).
+        For a full list see: https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task
+
+        :return: The generated responses from the model as a list of strings.
+        """
+        prompt = kwargs.get("prompt")
+        if not prompt:
+            raise ValueError(
+                f"No prompt provided. Model {self.model_name_or_path} requires prompt."
+                f"Make sure to provide prompt in kwargs."
+            )
+
+        stop_words = kwargs.pop("stop_words", None) or []
+        kwargs_with_defaults = self.model_input_kwargs
+        kwargs_with_defaults.update(kwargs)
+
+        # see https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task
+        params = {
```

Comment:
Have you thought about raising a warning if a kwarg is provided that doesn't match any of the expected parameters? Otherwise it's swallowed silently and typos in kwargs are hard to debug.

Reply:
Probably in the next iteration of invocation layers when we have stricter validation of kwargs in all layers :-)